src/main/java/pl/lukakan/showstracker/show/MovieService.java

```diff
@@ -0,0 +1,58 @@
+package pl.lukakan.showstracker.show;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.lukakan.showstracker.cast.role.model.Role;
+import pl.lukakan.showstracker.show.model.Genre;
+import pl.lukakan.showstracker.show.model.Movie;
+import pl.lukakan.showstracker.show.repository.GenreRepository;
+import pl.lukakan.showstracker.show.repository.MovieRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class MovieService {
+    private final MovieRepository movieRepository;
+    private final GenreRepository genreRepository;
+
+    @Autowired
+    public MovieService(MovieRepository movieRepository, GenreRepository genreRepository) {
+        this.movieRepository = movieRepository;
+        this.genreRepository = genreRepository;
+    }
+
+    public Optional<Movie> getById(Long id) {
+        return movieRepository.findById(id);
+    }
+
+    public void add(Movie movie) {
+        movie.getGenres().forEach(genre -> genre.removeMovie(movie));
```

Comment:
Zgaduję, że najpierw usuwasz, a potem dodajesz po to, żeby mieć tylko unikalne filmy. Zamiast tych dwóch operacji, to możesz zmienić, że Genre nie ma List<Movie> tylko Set<Movie> i Set przechowuje tylko unikalne wartości